### Inferences About the Difference Between Two Population Means: σ1 and σ2 Known

During the 2003 season, Major League Baseball took steps to speed up the play of baseball games in order to maintain fan interest (CNN Headline News, September 30, 2003).
The following results come from a sample of 60 games played during the summer of 2002 and a sample of 50 games played during the summer of 2003. The sample mean shows the
mean duration of the games included in each sample.

      2002 Season                             2003 Season
      n1 = 60                                 n2 = 50 
      Sample mean x1 = 172 min                Sample mean x2 = 166 min

a. A research hypothesis was that the steps taken during the 2003 season would reduce
the population mean duration of baseball games. Formulate the null and alternative
hypotheses.

b. What is the point estimate of the reduction in the mean duration of games during the
2003 season?

c. Historical data indicate a population standard deviation of 12 minutes is a reasonable
assumption for both years. Conduct the hypothesis test and report the p-value. At a .05
level of significance, what is your conclusion?

d. Provide a 95% confidence interval estimate of the reduction in the mean duration of
games during the 2003 season.

    Ans (a) Null Hypothesis, Ho = mu1-mu2 = 0, mu1 = mu2, The steps taken during the 2003 season did not reduce the population mean duration of baseball games.
    
            Alternative Hypothesis, Ha = mu1 > mu2, The steps taken during the 2003 season reduced the population mean duration of baseball games.

            So, we need to use Upper Tail Test

In [56]:
# Ans(b) 
#Point Estimate = x1 - x2 
Point_Estimate = 172-166
Point_Estimate

6

    Ans(b) Point estimate of the reduction in the mean duration of games = 6 min

In [57]:
# Import Libraries
import pandas as pd
import numpy as np
import math
from scipy import stats
from scipy.stats import norm

In [58]:
# Define a function to calculate Z and P value
def Z_and_p(x1, x2, Sigma1, Sigma2, n1, n2):
    z = (x1-x2)/(math.sqrt(((Sigma1**2)/n1)+ ((Sigma2**2)/n2)))
    if(z < 0):
        p = stats.norm.cdf(z)
    else:
        p = 1-stats.norm.cdf(z)
    print(z,p)

In [59]:
# Ans(c) Calculate value for Z and p for given value
# x1 = 172 min, x2 = 166 min, Sigma1 = Sigma2 = 12 min, n1 = 60, n2 = 50
Z_and_p(172,166,12,12,60,50)

2.611164839335468 0.00451171940904016


    Ans(c) Z = 2.61
           p = 0.0045

    Since, p value is less than the Level of Significance (0.05), we reject the null hypothesis and conclude that the steps taken during the 2003 season reduced the population mean duration of baseball games.

In [62]:
# Define a function to calculate Confidence Interval

def Confidence_Interval(x1, x2, sigma, n1, n2, alpha):
   
    # Calculate the standard error of the difference between the means
    std_err = math.sqrt(sigma**2 * (1/n1 + 1/n2))
    
    # Calculate the z-value for the given significance level
    z = norm.ppf(1 - alpha/2)
    
    # Calculate the margin of error
    margin_of_error = z * std_err
    
    # Calculate the lower and upper bounds of the confidence interval
    lower = (x1 - x2) - margin_of_error
    upper = (x1 - x2) + margin_of_error
    
    return (lower, upper)


In [63]:
# Ans(d) Calculate value for Confidence Interval
# x1 = 172 min, x2 = 166 min, Sigma = 12 min, n1 = 60, n2 = 50, alpha = 0.05
Confidence_Interval(172,166,12,60,50,0.05)

(1.4963456423405468, 10.503654357659453)

    Ans(d) Lower Limit = 1.49
           Upper Limit = 10.50

### Inferences About the Difference Between Two Population Means: σ1 and σ2 Unknown and Assume they are equal

Two catalyst are being analyzed to determine how they affect the mean yield of a chemical process. Specifically, catalyst 1 is currently in use, but catalyst 2 is acceptable.
Since catalyst 2 is cheaper, it should be adopted, Providing it does not change the Proceess yield.
A test is run in the plant and results are shown below.
Is there any difference between the mean yields?
Use 0.05, and assume equal variances

    Observation are :
    Catalyst 1 = [91.50, 94.18, 92.18, 95.39, 91.79, 89.07, 94.72, 89.21]
    Catalyst 2 = [89.19, 90.95, 90.46, 93.21, 97.19, 97.04, 91.07, 92.75]

    


Solution:

Hypothesis Test : 
The Pameter of Interest are mu1 and mu2, the mean process yield using catalysts 1 and 2, repectively.

Null Hypothesis = mu1 - mu2 = 0, mu1 = mu2

Alternative Hypothesis = mu != mu2

alpha = 0.05


In [3]:
import scipy.stats as stats

In [2]:
Catalyst_1 = [91.50, 94.18, 92.18, 95.39, 91.79, 89.07, 94.72, 89.21]
Catalyst_2 = [89.19, 90.95, 90.46, 93.21, 97.19, 97.04, 91.07, 92.75]

In [4]:
stats.ttest_ind(Catalyst_1, Catalyst_2, equal_var = True)

Ttest_indResult(statistic=-0.3535908643461798, pvalue=0.7289136186068217)

    t-test = -0.35
    p-value = 0.72
    Conclusion : As p-value is more than alpha value, Null hypothesis cannot be rejected.
    Therefore, we do not have strong evidence to conclude that catalyst 2 results in a mean yield that differs from the mean yield when catalyst 1 is used.
    
